src/math/Box3.js

```diff
@@ -435,20 +435,62 @@ class Box3 {
 
 	}
 
+	getVertexPoints( points ) {
+
+		if ( this.isEmpty() ) {
+
+			_points[ 0 ].set( 0, 0, 0 );
+			_points[ 1 ].set( 0, 0, 0 );
+			_points[ 2 ].set( 0, 0, 0 );
+			_points[ 3 ].set( 0, 0, 0 );
+			_points[ 4 ].set( 0, 0, 0 );
+			_points[ 5 ].set( 0, 0, 0 );
+			_points[ 6 ].set( 0, 0, 0 );
+			_points[ 7 ].set( 0, 0, 0 );
+
+		} else {
+
+			// NOTE: I am using a binary pattern to specify all 2^3 combinations below
+			_points[ 0 ].set( this.min.x, this.min.y, this.min.z ); // 000
+			_points[ 1 ].set( this.min.x, this.min.y, this.max.z ); // 001
+			_points[ 2 ].set( this.min.x, this.max.y, this.min.z ); // 010
+			_points[ 3 ].set( this.min.x, this.max.y, this.max.z ); // 011
+			_points[ 4 ].set( this.max.x, this.min.y, this.min.z ); // 100
+			_points[ 5 ].set( this.max.x, this.min.y, this.max.z ); // 101
+			_points[ 6 ].set( this.max.x, this.max.y, this.min.z ); // 110
+			_points[ 7 ].set( this.max.x, this.max.y, this.max.z ); // 111
+
+		}
+
+		if ( _points !== points ) {
+
+			points = points || [];
+			points.length = 0;
+
+			for ( let i = 0; i < _points.length; i ++ ) {
+
+				points[ i ] = _points[ i ].clone();
```

Comment:
It is the app's responsibility to allocate memory, so cloning is not permissable.

Reply:
If cloning is not used, the value of 'points' cannot be guaranteed to be fixed externally.



Additionally, this is only a personal requirement for me and I am not sure if it is necessary to actually merge it into the main branch.

If it's not suitable, then I'll close this PR.

Reply:
Yes, it may be more appropriate as an app-level method. @Mugen87 will make that decision.

Reply:
First of all, thanks for all the work that has been done so far.

But given how `getVertexPoints()` works, it's probably better to leave the method out for now. If more users request such a feature, we can reconsider the PR.

Reply:
I also agree.